List View: add an indicator of when a position type is set for a block

andrewserong opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Part of https://github.com/WordPress/gutenberg/issues/47043

Add an indicator to the block icon in the list view to flag a position type when one is set.

This follows on from feedback in https://github.com/WordPress/gutenberg/pull/49085#issuecomment-1469711661

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

When a user sets a Group to sticky, it would be helpful for the sticky state to be visible from within the list view.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

* Add a function to get a position type label based on a block's attributes.
* Update the `useBlockDisplayInformation` hook to retrieve and pass along the position label and position type.
* Add a `pinSmall` icon.
* When a block's position is set to `sticky`, display a pin icon next to where the lock icon is displayed.

<!--
* Wrap the list view's block icon in a Tooltip with the text `Position: Sticky` when a position label is available
* Add positionSticky and positionAbsolute icons
* Layer the positionSticky icon over the block icon when `sticky` is set on the block -->

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

1. With TT3 active (or any blocks theme with `appearanceTools` set to true), add a Group block at the root level of a post, or at the root level of a template.
2. In the sidebar Position panel, select "Sticky".
3. In the List view, you should see a pin icon flagging that the block is sticky.

## Screenshots or screencast <!-- if applicable -->

<img width="347" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/2fd5e59b-50a5-49ab-b3d7-1f35b1fe4b16">

<!--

| Screenshot | Screengrab |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/14988353/227091694-2a3fd2d4-0d35-420e-8418-49004425e628.png" width="350" /> | ![2023-03-24 10 21 05](https://user-images.githubusercontent.com/14988353/227386636-fabe111a-54c8-44ae-bb5c-4258e9b6629a.gif) |

-->

<!--

![image](https://user-images.githubusercontent.com/14988353/225500630-6d755dcc-99bf-4ef9-b84f-4006e0db4a1c.png)

## Other potential ideas

I tried out a few ideas, but this one seemed the simplest to implement while still being clear in the UI. Some other potential ideas could be to put the text in a pill, or try out an icon. These ideas had some drawbacks:

| Use an icon (the pin doesn't seem clear to me as to what it's representing, also we'll need to support text labels anyway) | Display as a pill (it's no longer clear what Sticky is compared to the anchor id if one exists) |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/225500961-b0a44c33-3b83-4c2d-9f03-a0385713c78e.png) | ![image](https://user-images.githubusercontent.com/14988353/225502121-3a113730-ba41-4afb-8def-2c88ad43108c.png) |
-->

andrewserong committed:
List View: add a position label when a position type is set for a block

alexstine commented:
Before this gets merged, lets do figure out where this info might be useful for screen readers. 👍

andrewserong commented:
> Before this gets merged, lets do figure out where this info might be useful for screen readers. 👍

Absolutely! Happy for any feedback or ideas on how best to word the label for screen readers.

andrewserong committed:
Move get label function to the useBlockDisplayInformation file

jasmussen commented:
It definitely becomes tricky when there's both anchor and sticky chips. But that treatment still seems the most compelling to me, outside of a better icon.

Question: would it be inappropriate for the "Sticky" chip to take precedence if the list view contains both a sticky chip, and an anchor chip? The latter feels more quality of life to me, and is also surfaced in the inspector.

Alternatively, is it possible to hide these chips, one by one, as the title of the block grows longer, or nesting grows deeper? This is probably something to consider separately.

richtabor commented:
> Question: would it be inappropriate for the "Sticky" chip to take precedence if the list view contains both a sticky chip, and an anchor chip? The latter feels more quality of life to me, and is also surfaced in the inspector.

I'm not sold we need a sticky chip. 

Why should sticky be called out here, when nearly all other attributes for the group aren't? 

I'm more in favor of adapting the block icon to indicate stickiness, if we need/should do this.

andrewserong commented:
Thanks for the feedback and ideas, folks!

> It definitely becomes tricky when there's both anchor and sticky chips. But that treatment still seems the most compelling to me, outside of a better icon.
> Question: would it be inappropriate for the "Sticky" chip to take precedence if the list view contains both a sticky chip, and an anchor chip?

I agree — I think it's the clearest indicator visually, and when I attempted to get both the anchor and the sticky chip/pill to be on the same row, I very quickly ran into styling issues with the limited real estate we have there. If we agree on a sticky chip, I think the most pragmatic option is to hide the anchor if it exists, and like you mention, we can always look at seeing if we can get them both displayed in a follow-up if need be.

> Why should sticky be called out here, when nearly all other attributes for the group aren't?

That's a good question — I think the position type is worth calling out within the list view because unlike other block supports / styling settings, position affects where the block appears within the page. In a sense it breaks out of its position within the flow of the document, so I think it'd be good to find ways that make it easier for the user to get back to that element or select it. Also, there'll likely be very few elements on the page that are sticky, so it hopefully won't result in too much visual clutter.

> I'm more in favor of adapting the block icon to indicate stickiness, if we need/should do this.

The main drawbacks I can see for adjusting the block icon are:

* Sticky positioning will likely be available on at least a few different block types in the future (e.g. cover block, headings, potentially), so whatever visual indicator we use will need to be decoupled from any one particular block type. So, if we switch out the icon, does the icon we switch it for make sense for any arbitrary block that might be sticky?
* Although this isn't a button, for folks who use the "Show button text labels" option in the editor, the expectation is that the UI deprioritises the use of icons to communicate information, so for this label, I think it'd be good to still have a text alternative, which probably leans us toward a pill idea.

---

I've added the "Needs Design" label — I'm happy to have a go at implementing whichever idea we think will work best here, but since it's a little fiddly to work with, if anyone's free to put together a mockup of how it'd ideally look, that would be very helpful! Some ideas:

* As Joen mentioned — what if we go with the sticky chip idea, but hide the id attribute if present, so the sticky chip will take precedence
* If a heading with an id attribute is adjacent to the sticky group within the list view, they'll both have chips on them. Should the sticky chip be a different colour / look any different to distinguish it visually? E.g. different background colour or uppercase?

I'll keep chipping away at this PR next week!

jasmussen commented:
Some of the motivation is the lack of an immediate visual effect when toggling sticky. But to your point, which is a good one, there are other ways to do this. 

I do like your idea of an icon change. Figma has this toggle for abs position:

<img width="53" alt="Screenshot 2023-03-17 at 08 19 03" src="https://user-images.githubusercontent.com/1204802/225838557-1682c95f-c05e-4036-9979-812c6cd59e38.png">

In the layers panel, the icon then gets those 4 corners added. In this example, a symbol instance before and after stickied:

<img width="230" alt="Screenshot 2023-03-17 at 08 19 15" src="https://user-images.githubusercontent.com/1204802/225838654-3ba9a278-3057-45a3-9671-2a92ea79e8ab.png">
<img width="227" alt="Screenshot 2023-03-17 at 08 19 10" src="https://user-images.githubusercontent.com/1204802/225838666-99fc75cf-5c08-4956-a54d-b3da67fc1b53.png">

That does seem less intrusive, and perhaps simpler to implement?

Marc-pi commented:
I've read the other associated issues /iteration as well.

IMHO the main stakes, **from an end user /editorial point of view**, are : 
- to keep the purple color in case of partial template, it is important
- not to transform too much the hightlight of the header zone (it is a major structural part of the templates) 
Now with the group encapsulation, it's different, we loose in a sense the information (header is behind a group)
"If i've set a heading zone, it's strange to see my work being encapsulated just after having activated a sticky option" :  it is a sort of hijacking :-)
Is there a way to improve that part, i.e move group below inside ? 
there is  indeed a need to make the change more transparent / less intrusive
- the icon idea is to be kept, maybe add a tooltip on a:hover it

here is what i can see on some of my websites with Senff's plugin (https://wordpress.org/plugins/sticky-block/), just to compare
![image](https://user-images.githubusercontent.com/919332/225848636-740b50ac-7295-4415-9b6b-5d5bf10f1c94.png)

Would be nice to check what other builders do.

andrewserong committed:
Try using icon over the block icon instead of a text label to indicate sticky status

andrewserong commented:
Thanks for the feedback and ideas, folks! I've had a go at trying @jasmussen's idea of adding an indicator on top of the block icon. I don't mind it, but I'm also a little torn — it does add an indicator, but it also looks a little busy to my eyes, and isn't clear to me what it does. But perhaps it's clearer from context when a user interacts with it. Here's screenshot of the current state of this PR:

<img width="347" alt="image" src="https://user-images.githubusercontent.com/14988353/226261531-e9369f17-9d2e-475d-bab2-d420b4bf5a40.png">

And a screengrab of switching position types:

![2023-03-20 17 16 45](https://user-images.githubusercontent.com/14988353/226261800-a05b9192-d3cd-4af5-8595-43e3d599b209.gif)

If people like this approach, then I think the next steps would be:

* Is there an svg icon we should use instead of the background-image data URLs I hacked in there (I added them using this tool: https://kovart.github.io/dashed-border-generator/)? The `fullscreen` icon is fairly close, but it's a bit too thick to work with the existing block icons. Alternately, we can tweak the data URLs as needed to get the right sizing.
* We'll then need to figure out a good way to make it accessible. I like the idea of trying out a Tooltip and ensuring there are accessible labels to flag the sticky status.

Thanks again for the collaboration!

jasmussen commented:
Thanks for trying that. Outside some treatment details, I think it's interesting. What do you think, @WordPress/gutenberg-design

jameskoster commented:
It seems plausible that we might want to indicate other position properties like `fixed` and `absolute` in List View. Perhaps it's my bias from Figma, but this particular treatment suggests `absolute` positioning to me, so I'd welcome alternative ideas. 

Maybe just a single stroke at the top of the svg?

<img width="350" alt="Screenshot 2023-03-20 at 16 20 32" src="https://user-images.githubusercontent.com/846565/226403783-461fc0e9-cf06-4fa5-ba81-7269d6d81b33.png">

> and isn't clear to me what it does

I feel this too. It would be nice to avoid additional text, but it doesn't seem like the end of the world to me, especially as (I assume) we'll at least need a VisuallyHidden label for a11y.

apeatling commented:
> and isn't clear to me what it does
> I feel this too. It would be nice to avoid additional text, but it doesn't seem like the end of the world to me, especially as (I assume) we'll at least need a VisuallyHidden label for a11y.

A big +1 from me for using something that makes it immediately obvious what it means. I agree that additional text here is not the end of the world and would improve clarity in a UI that is designed to improve clarity of the entire document structure.

andrewserong commented:
Yes, I think I'm leaning toward returning to text display of the position status, too. If we switch this PR back to using text to indicate the position type, do we have a preference for:

* Augment the block display title (e.g. as this PR originally was, so it'd say `Group (sticky)`, or:
* Use a `sticky` chip/pill similar to the styling used by the id attribute?

jameskoster commented:
I'm not quite ready to give up on the icon approach, mostly because when we consider anchors, absolute positioning, and nesting, space in the row is really at a premium.

Would it be worth trying something like this:

<img width="300" alt="Screenshot 2023-03-21 at 13 58 21" src="https://user-images.githubusercontent.com/846565/226628956-17144163-66b7-49b1-bcfe-afd6eae7ca1d.png">

andrewserong commented:
> Would it be worth trying something like this:

Definitely worth trying! I'm happy to give it a go.

andrewserong committed:
Try more subtle icon / styling idea, and add in a tooltip

andrewserong commented:
Update: I think that direction improves things a fair bit, particularly adding in the tooltip. I also like how the grey in that mockup looks a bit more subtle, indicating that it's a status indicator of some kind as opposed to part of the block icon. Here's how this PR is looking now with a simple top-border (I'd love to try out the icon once we've got some icons to play with):

| Screenshot | Screengrab with Tooltip |
| --- | --- |
| <img width="346" alt="image" src="https://user-images.githubusercontent.com/14988353/226807404-4af776ae-18e9-4941-851d-b36b127568b6.png"> | ![2023-03-22 16 08 39](https://user-images.githubusercontent.com/14988353/226807587-01c36259-2119-4cac-8968-7ccf9401c99a.gif) |

@jameskoster if this direction still looks like a good one to keep exploring, do you mind giving me some svg markup from your mockup to try for the sticky icon? It's that tiny bit more complex than the top border color, so I figured an svg might work. Also some more detailed / polishing questions:

* Which colour should we use for the icon? (I've gone with `$gray-600`)
* Which colour should it be when the block is selected? (I've gone with `$white`)
* Which colour should it be when the mouse hovers over the block? Currently it stays as gray for now.

jasmussen commented:
I don't dislike that, mainly because it's managing the prominence of the feature. 

I also recall an older mockup that bunched sticky at the top, but with a heading separator. I guess that doesn't consider nested blocks?

jameskoster commented:
Thanks @andrewserong. Here are the SVGs:

**Sticky**
```
<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M24 0V4H23V1H1V4H0V0H24Z"/>
</svg>

```

**Absolute**
```
<svg fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
    <path d="M4 0H0v4h1V1h3V0Zm20 4V0h-4v1h3v3h1Zm0 20h-4v-1h3v-3h1v4ZM0 20v4h4v-1H1v-3H0Z" />
</svg>
```

I used `$gray-500` in the mockup.

> Which colour should it be when the mouse hovers over the block?

Probably fine for it to adopt the accent color, same as it adopts the white fill on selection.

> I also recall an older mockup that bunched sticky at the top, but with a heading separator

Perhaps you were thinking about Figma's handling of fixed elements? 

<img width="284" alt="Screenshot 2023-03-22 at 10 59 09" src="https://user-images.githubusercontent.com/846565/226883559-121624d0-2c40-4240-8a3c-58a4b3423f4b.png">


I think it works for `fixed` because that position abstracts the element from its siblings in terms of structure/order. It's not quite the same for `sticky` so I don't think it would work in that context.

andrewserong committed:
Try provided svg icons

andrewserong commented:
Thanks @jameskoster! I've updated to use the real icons. Here's how it's looking now:

![2023-03-23 13 25 29](https://user-images.githubusercontent.com/14988353/227093377-4ce5d5f2-bb76-4633-9c62-a29a45dc3626.gif)

A couple of details:

* It seems that `$gray-500` doesn't exist in the list of variables. From a quick read of [this older PR](https://github.com/WordPress/gutenberg/pull/23648), it sounds like it might have been excluded to ensure a restricted palette? I've gone with `$gray-400` for now, but I can switch it back to `$gray-600` if the contrast is too low, now.
* I've updated it so that the accent color is used when hovering, and it's white when selected.
* I've included the icon for absolute in this PR, so that it's easy for us to use in follow-ups, but haven't added any logic to render it yet, as we haven't implemented that feature yet.

The more I play with it, the more this approach is growing on me. I quite like it!

Let me know if anyone would like any other changes, I think this PR's pretty close to review-ready now.

andrewserong commented:
> Before this gets merged, lets do figure out where this info might be useful for screen readers. 👍

@alexstine this PR has shifted direction a little, so it now adds an indicator icon over the block icon in the list view. In thinking about how we can flag the position status for screen readers, I think we might want to do that after #48461 lands, since the button in the list view currently has `aria-hidden` set. Let me know if you think that's a blocker for this PR. From my perspective, I thought we could possibly land this PR first if it winds up being approved, and look at updating the `aria-label` to indicate sticky position in a follow-up once #48461 is in.

alexstine commented:
@andrewserong Yes, I think this is okay. I will try to find some time to continue work on the refactor. Might even be able to rely on `aria-label` less if I can make the components themselves accessible. We'll see.

richtabor commented:
I'm not 100% for it, but I don't know of a better indicator. 

Perhaps it shouldn't change color on hover though? The gray -> blue is much more of a visual change than the darker icon color, leading to a bit too much attention. 

https://user-images.githubusercontent.com/1813435/227242035-4b2ab7a9-b1b8-4fbc-a25c-0cedb370512d.mp4

jameskoster commented:
> Perhaps it shouldn't change color on hover though? The gray -> blue is much more of a visual change than the darker icon color, leading to a bit too much attention.

Fair point!

andrewserong committed:
Don't change color when hovered

andrewserong commented:
Yes, good point!
I've updated it so that it only switches color when the block is selected. So on hover, the color now stays `$gray-400`:

![2023-03-24 10 21 05](https://user-images.githubusercontent.com/14988353/227386748-0656f17f-f6c2-43d3-a692-e011966a0c7d.gif)

tellthemachines commented on packages/block-editor/src/components/list-view/block-select-button.js:
```diff
@@ -79,7 +94,23 @@ function ListViewBlockSelectButton(
 				<ListViewExpander onClick={ onToggleExpanded } />
-				<BlockIcon icon={ blockInformation?.icon } showColors />
+				{ positionLabel ? (
+					<Tooltip text={ positionLabel }>
```
Could the tooltip instead be a child of the whole focusable list view item? I think that will allow it to display when the item is focused too. As it is, it only displays on hover so is not keyboard accessible.

andrewserong replied:
Oh, good question. Would that then make the Tooltip too prominent for users moving up and down through the list view? Currently when you're within a list view item, you can press the right arrow to focus on the settings dropdown button, and it might look odd if you're switching between tooltips 🤔 

<img width="416" alt="image" src="https://user-images.githubusercontent.com/14988353/227428156-96595f2d-a7c1-4ad6-bc7c-d884883b13fb.png">

Happy to try it out on Monday, though!

tellthemachines replied:
Good point, we'd have to try it and see. Assuming the position tooltip would only show when a block is sticky/fixed, it shouldn't be too disruptive?

andrewserong replied:
Looking at this again, I'm not sure if we can get the Tooltip on the root element here (so, displaying when the button is focused) without the work in #48461 as it appears we'd need to set the Tooltip at the Button component level, which is currently set to `aria-hidden`. Enabling the tooltip there would likely require #48461 so that we can use the `aria-label`. The solution then might be to augment the `ariaLabel` to include the position status, and we could look at ensuring we deal with keyboard accessibility at the same time as accessibility for screen readers.

What do you think? I don't think there's any particular urgency for this PR, so we could always wait for #48461 if we think displaying the tooltip for users navigating by keyboard is a blocker 🙂

tellthemachines replied:
If it's ok to wait, that might be best. We should ensure tooltips are always visible on focus.

andrewserong replied:
Sounds good 👍 I'll see what I can do to help out with that other PR.

tellthemachines reviewed (commented):
Just one question below regarding the Tooltip, but code otherwise looks good!

apeatling commented:
The indicator is looking much better, I could mostly infer what it means assuming I made the initial change. The tooltip makes sense to help those that it might not make sense for initially. 👍

mtias commented:
Can we try a plain icon next to the ellipsis menu like we communicate "locked" state? This seems very ad hoc

andrewserong commented:
> Can we try a plain icon next to the ellipsis menu like we communicate "locked" state? This seems very ad hoc

Sure, I really like that idea. I think earlier feedback from @richtabor in https://github.com/WordPress/gutenberg/pull/49122#issuecomment-1472753132 indicated trying to make it less prominent, but that was when the discussion was about a text-based chip rather than a separate icon.

If any designers have an icon to share that I can play with, I can put it together!

jameskoster commented:
The challenge has always been coming up with a compelling icon to indicate sticky positioning. Iirc this is why the text label was originally suggested.

There are metaphors like pins / map markers, anchors, magnets... but for one reason or another they don't feel quite right. Trying to communicate the behavior more literally is very tricky in such a small space.

mtias commented:
@jameskoster I think it's more about adding a marker so the block stands out. Once you associate it to the behavior it flows more easily.

jameskoster commented:
One option could be a "note" affordance on each item. It could contain details like position, anchor, behavior.

<img width="895" alt="Screenshot 2023-06-06 at 13 50 48" src="https://github.com/WordPress/gutenberg/assets/846565/9dae8de0-31bc-4911-8980-745cfe970317">

Another could be a different treatment of the row itself:

<img width="860" alt="Screenshot 2023-06-06 at 14 36 47" src="https://github.com/WordPress/gutenberg/assets/846565/a70741a0-8ebb-4ee5-9b0f-78a64b1ddb30">


An icon next to the ellipsis / lock can also work, but I'm not really sure which icon to use :)

<img width="1260" alt="Screenshot 2023-06-06 at 14 35 51" src="https://github.com/WordPress/gutenberg/assets/846565/e8a45c02-257c-4d2b-9ab5-edf95a9ec94a">

The pin feels most obvious, but it's used elsewhere in WordPress with a totally different meaning.

Marc-pi commented:
personaly i vote for a strict vertical pin ;-)
as per https://github.com/WordPress/gutenberg/pull/49122#issuecomment-1473369926

andrewserong commented:
The pin icon next to the lock icon looks like a good approach to try to me! I'll have a go at that next week unless there are any other ideas in the meantime.

Jabe64 commented:
Great work! Another idea would be a mix of the lock icon and a vertical or horizontal line around it depending on which part is sticky.

For example:
- a horizontal line above the lock icon whenever it's sticky to the top
- a horizontal line below the lock icon whenever it's sticky to the bottom
- a vertical line on the left of the lock icon whenever it's sticky to the left
- a vertical line on the right of the lock icon whenever it's sticky to the right

And versions for mix of positions for example: sticky to both right and bottom.

Other sticky positions aren't available yet (it’s only sticky to the top if I’m not wrong) and still need to be considered in terms of UI as listed in #47043:
>Look into position UI for top/left/right/bottom positions and offsets, in particular support for fixed/sticky footers

But it might be a good thing to take those future options in consideration for creating the proper icon(s).

andrewserong committed:
Try separate sticky icon instead

andrewserong commented:
Thanks for the feedback, all! I think for simplicity sake, trying out a sticky icon that is similar in size to the lock icon could be good, and possibly a pragmatic path forwards.

I've updated this PR to use the existing `pin` icon as a placeholder, and here's how it's currently looking (this example just puts two sticky blocks next to each other so you can see the sticky icon in its selected and deselected state, and how it looks with a lock icon next to it):

<img src="https://github.com/WordPress/gutenberg/assets/14988353/882e8a60-0460-4d19-9c6e-3e8e4a5c8111" width="350" />

If this direction seems okay, then I think the next step would be for us to add a `pinSmall` icon, a bit like how we have the `lockSmall` icon, so that the icons can be the same size as in the mockups in https://github.com/WordPress/gutenberg/pull/49122#issuecomment-1578787833. @jameskoster — if you think it's worth pursuing, do you have some `SVG` markup I can use to add in an extra icon?

Marc-pi commented:
@andrewserong i love this new iteration / icon, keep it up ! good job

jameskoster commented:
Thanks @andrewserong, I think we can slightly scale up the shape in the `post` icon. Here's an svg:

```
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10.9706 10.1586C10.2819 10.0621 9.56905 10.1784 8.93842 10.5075C8.88672 10.5345 8.83557 10.5629 8.78504 10.5928C8.54529 10.7344 8.3193 10.9082 8.11331 11.1142L9.83702 12.8372L7.00054 15.75L7 17H8.25064L11.1628 14.1632L12.8857 15.8865C13.0917 15.6805 13.2655 15.4545 13.4071 15.2147C13.437 15.1642 13.4654 15.1131 13.4924 15.0614C13.8215 14.4308 13.9378 13.718 13.8413 13.0293L17 10.6946L13.3053 7L10.9706 10.1586Z" />
</svg>

```

andrewserong committed:
Add pinSmall icon

andrewserong commented:
Thanks @jameskoster, that's looking good to me! I've added a `pinSmall` icon using that markup, here's a current screenshot:

<img width="347" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/2fd5e59b-50a5-49ab-b3d7-1f35b1fe4b16">

I think this should be ready for a final review now.

jameskoster committed:
Minify pinSmall

jameskoster reviewed (approved):
It's working well for me.

andrewserong committed:
Add additional optional chaining because attributes can technically be null

andrewserong commented:
Thanks for the review @jameskoster, and for minifying the SVG. I'll merge this in now.
